src/notebooks/controllers/vscodeNotebookController.ts

```diff
@@ -97,7 +97,7 @@ export class VSCodeNotebookController implements Disposable, IVSCodeNotebookCont
         selected: boolean;
         notebook: NotebookDocument;
     }>();
-    private pendingCellAddition: Promise<void> | undefined;
+    private pendingCellAdditions = new Map<NotebookDocument, Promise<void>>();
```

Comment:
Might want to change this to a weakmap
```suggestion
    private pendingCellAdditions = new WeakMap<NotebookDocument, Promise<void>>();
```